File: swift/sample/Runner/Custom/Callback.swift
import UIKit
import xplpc

class Callback {
    static func batteryLevel(message: Message, r: Response) {
        let level = 100

        let suffix: String = message.get("suffix") ?? ""

        if suffix.isEmpty {
            r("\(level)")
        } else {
            r("\(level)\(suffix)")
        }
    }
}

File: kotlin/sample/app/src/main/java/com/xplpc/runner/custom/Callback.kt
package com.xplpc.runner.custom

import android.content.Context.BATTERY_SERVICE
import android.os.BatteryManager
import com.xplpc.message.Message
import com.xplpc.message.Response
import com.xplpc.runner.app.Application

object Callback {
    fun batteryLevel(data: Message): Response {
        val bm = Application.instance.getSystemService(BATTERY_SERVICE) as BatteryManager
        val level = bm.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY)

        val suffix = data.value<String>("suffix")

        return if (suffix != null) {
            Response("$level$suffix")
        } else {
            Response("$level")
        }
    }
}
